src/include/duckdb/common/http_state.hpp

```diff
@@ -36,7 +38,7 @@ class HTTPState {
 	//! Mutex to lock when getting the cached file(Parallel Only)
 	mutex cached_files_mutex;
 	//! In case of fully downloading the file, the cached files of this query
-	unordered_map<string, CachedFile> cached_files;
+	unordered_map<string, shared_ptr<CachedFile>> cached_files;
```

Comment:
Perhaps we can clean up the code by moving the members to be private (`cached_files` for the `HTTPState`, and `capacity`/ `data`/`finished` for `CachedFile`) and moving accesses to them to GETers that obtain the lock correctly? As it stands it is still easy to accidentally cause a race condition by forgetting to grab the lock (and this might happen elsewhere in the code)

Reply:
Good point, did a little refactor to make it safer